runtime/util_core/j9shchelp.c

```diff
@@ -141,80 +141,67 @@ getShcModlevelForJCL(uintptr_t j2seVersion)
 {
 	uint32_t modLevel = 0;
 	switch (j2seVersion) {
-	case J2SE_15 :
-		modLevel = J9SH_MODLEVEL_JAVA5;
-		break;
-	case J2SE_16 :
-		modLevel = J9SH_MODLEVEL_JAVA6;
-		break;
-	case J2SE_17 :
-		modLevel = J9SH_MODLEVEL_JAVA7;
-		break;
 	case J2SE_18 :
 		modLevel = J9SH_MODLEVEL_JAVA8;
 		break;
 	case J2SE_19 :
 		modLevel = J9SH_MODLEVEL_JAVA9;
 		break;
-	case J2SE_V10 :
-		modLevel = J9SH_MODLEVEL_JAVA10;
-		break;
-	case J2SE_V11 :
-		modLevel = J9SH_MODLEVEL_JAVA11;
+	default: 
+		modLevel = JAVA_SPEC_VERSION_FROM_J2SE(j2seVersion);
 		break;
 	}
 	return modLevel;
 }
 
 /**
- * Return the j2se version for a given modification level
- * 
+ * Return the jave spec version for a given modification level
  * @param [in] modlevel  The modification level
  * 
  * @return the j2se version or 0 if one cannot be found
  */
 uint32_t
 getJCLForShcModlevel(uintptr_t modlevel)
 {
-	uint32_t j2seVersion = 0;
+	uint32_t javeVersion = 0;
 	switch (modlevel) {
 	case J9SH_MODLEVEL_JAVA5 :
-		j2seVersion = J2SE_15;
+		javeVersion = 5;
 		break;
 	case J9SH_MODLEVEL_JAVA6 :
-		j2seVersion = J2SE_16;
+		javeVersion = 6;
 		break;
 	case J9SH_MODLEVEL_JAVA7 :
-		j2seVersion = J2SE_17;
+		javeVersion = 7;
 		break;
 	case J9SH_MODLEVEL_JAVA8 :
-		j2seVersion = J2SE_18;
+		javeVersion = 8;
 		break;
 	case J9SH_MODLEVEL_JAVA9 :
-		j2seVersion = J2SE_19;
-		break;
-	case J9SH_MODLEVEL_JAVA10 :
-		j2seVersion = J2SE_V10;
+		javeVersion = 9;
 		break;
-	case J9SH_MODLEVEL_JAVA11 :
-		j2seVersion = J2SE_V11;
+	default:
+		if (modlevel > 9) {
+			/* J9SH_MODLEVEL_JAVA9 is 5. Does not have modlevel that is 6,7,8,9 */
+			javeVersion = modlevel;
+		}
```

Comment:
seems there should be an assert for the else case

Reply:
This function is used when iterating/destroying the shared caches. I guess it is fine to not assert in this cases

Reply:
Not sure what you mean. If modlevel 6 - 9 should never occur, its fine to assert if they do.

Reply:
Oh, you mean because jdk10 already uses 6, but is being converted to 10? Good point, how about adding a case for 6 to handle the transition.

Reply:
Yes, the current JDK10 is already using 6. 
Also it is possible that someone put a fake cache file whose name is C280M7F0A32P_Cache1_G35 (modLevel is 7), VM would exit on this assertion failure and generate core files. User might not want to see this when destroying the caches (either using CML or SharedClassUtilities API). 

BTW, the cache name length needs to be increased. We currently have only 1 digit for modlevel in the cache name. We need two digits now. I will fix this.